### PR TITLE
fix(android): Check in material-stepper as internal Maven dependency

### LINKED ISSUE #12260: chore(android): Could not find com.stepstone.stepper:material-stepper:4.3.1 on a new development installation
`android/build.sh`: 

```
A problem occurred configuring root project 'KMAPro'.
> Could not resolve all files for configuration ':classpath'.
   > Could not find com.stepstone.stepper:material-stepper:4.3.1.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/com/stepstone/stepper/material-stepper/4.3.1/material-stepper-4.3.1.pom
       - https://jcenter.bintray.com/com/stepstone/stepper/material-stepper/4.3.1/material-stepper-4.3.1.pom
       - https://repo.maven.apache.org/maven2/com/stepstone/stepper/material-stepper/4.3.1/material-stepper-4.3.1.pom
     Required by:
         project :
```

build.gradle:
```
        // From jcenter() which could be sunset in future
        // https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/
        classpath 'com.stepstone.stepper:material-stepper:4.3.1'
```

So, yes, this relates to JCenter shutdown which happened last week (good timing huh?)

> UPDATE 7/2/2024: After over three years of maintaining JCenter as a read-only repository for legacy Java packages, JFrog is moving forward with an official sunset of JCenter, and, in coordination with Maven Central, will begin automatically redirecting all JCenter requests to the Maven Central Repository. The first of several service brown-outs is scheduled for July 30th and will periodically continue until the final shutdown of JCenter on August 15th. For more details, please see [JFrog’s official announcement](https://jfrog.com/blog/jcenter-sunset/).

We should either replace this component with a newer one, or bring the component into the repository.

### PR DESCRIPTION
**Pull request by darcywong00**
Fixes #12260

JFrog recently retired the JCenter maven repository and it no longer provides the (out of maintenance) material-stepper dependency. Keyman uses the stepper in the multi-language picker UI during keyboard installation.

This PR follows the guide on [setting up a local maven repository](https://brightinventions.pl/blog/migrating-away-from-bintray-jcenter-when-there-is-no-alternative-repository/#setting-up-a-local-maven-repository) to reference the checked-in dependencies.

@mcdurdin - can you see if this branch builds on a new CI? (that doesn't have the Gradle cache of the material-stepper files)

## User Testing
**Setup** - Install the PR build of Keyman for Android on a device/emulator

**TEST_STEPPER** - Verifies Keyman can use the stepper in choosing language during keyboard installation
1. Launch Keyman for Android and dismiss the "Get Started" menu
2. From Keyman settings --> Install the Keyman keyboard for "sil_cameroon_qwerty".
3. During the keyboard installation, verify you can choose language(s) to install with the sil_cameroon_qwerty keyboard.
4. When the keyboard installation finishes, verify the Keyman keyboard is now sil_cameroon_qwerty

### CODE REVIEW

**Comment by darcywong00** [comment]
Might need to :cherries: pick to stable-17.0 if building on a new CI

File: android/KMAPro/build.gradle [hunk 1]
```diff
@@ -13,14 +17,15 @@ buildscript {
 
         // From jcenter() which could be sunset in future
         // https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/
-        classpath 'com.stepstone.stepper:material-stepper:4.3.1'
+        classpath 'com.stepstone.stepper:material-stepper:4.3.1@aar'
```
Comment (jahorton) [inline android/KMAPro/build.gradle:20]: Line 14 (old) / 18 (new):

> which could be

... so... "has been", now?  Looks like the comment needs an update of some sort.

**Comment by mcdurdin** [comment]
> @mcdurdin - can you see if this branch builds on a new CI? (that doesn't have the Gradle cache of the material-stepper files)

I am testing this locally, fighting mismatching Java versions currently.

As far as I understand, using `mavenLocal()` is not recommended:
* https://docs.gradle.org/current/userguide/declaring_repositories_adv.html#sec:case-for-maven-local

As an alternative, is there a replacement stepper component we could use?

**Comment by dinakaranr** [comment]
# Test Results
I tested this issue with the attached "Keyman 18.0.95-alpha-test-12242" build on the Android 14 physical device. I'm sharing my observation here.

* **TEST_STEPPER (passed):** 
1. Installed the "Keyman-18.0.96.apk" file and gave all permissions to the application.
2. Checked the "Enable Keyman as system-wide keyboard" and set the keyboard as the default keyboard box on the settings page.
3. I selected the "SIL EuroLatin" default keyboard.
4. Open the Keyman app. Enable the "Predictions" to install the "Dictionary."
5. Open the "Download" keyboard by clicking settings/Install Keyboard or Dictionary.
6. Install the "sil_cameroon_qwerty" keyboard. 
7. Select the language on the "Select language for Cameroon QWERTY" dialog while installing the keyboard.
8. Open the "Keyboard" dialog by pressing the globe key.
9. Verified the installed keyboard had a "Cameroon QWERTY" appear below the language name.
It works great for me. Thank you.

**Comment by darcywong00** [comment]
> As far as I understand, using `mavenLocal()` is not recommended:
> 
> * https://docs.gradle.org/current/userguide/declaring_repositories_adv.html#sec:case-for-maven-local
> 
> As an alternative, is there a replacement stepper component we could use?

The build.gradle has the path for the checked in stepper, so I was also able to remove `mavenLocal()`

File: android/KMAPro/build.gradle [hunk 2]
```diff
@@ -2,25 +2,29 @@
 buildscript {
     repositories {
         google()
-        jcenter()
+        //jcenter() deprecated August 2024
```
Comment (mcdurdin) [inline android/KMAPro/build.gradle:5]:
```suggestion
```

We can use git history to see more detail, I reckon just remove this line

File: android/KMAPro/build.gradle [hunk 3]
```diff
@@ -2,25 +2,29 @@
...
+        classpath 'com.stepstone.stepper:material-stepper:4.3.1@aar'
     }
 }
 
 allprojects {
     repositories {
+        maven { url uri("${projectDir}/libs") }
         google()
-        jcenter()
+        //jcenter() deprecated August 2024
```
Comment (mcdurdin) [inline android/KMAPro/build.gradle:27]:
```suggestion
```

File: android/KMAPro/kMAPro/build.gradle [hunk 4]
```diff
@@ -145,7 +145,7 @@ repositories {
         dirs 'libs'
     }
     google()
-    jcenter()
+    //jcenter()
```
Comment (mcdurdin) [inline android/KMAPro/kMAPro/build.gradle:148]:
```suggestion
```

**Review by mcdurdin — APPROVED** [review]

**Comment by mcdurdin** [comment]
Note: I still haven't had a chance to test on my devbox, perhaps hold off merge until I have done that

**Comment by mcdurdin** [comment]
Yes, this works on my machine. We should go ahead and cherry-pick to stable-17.0 too

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.101-alpha](https://keyman.com/downloads/releases/alpha/18.0.101)

**Pull request merged**